packages/headless/src/controllers/result-list/headless-result-selection-helpers.ts

```diff
@@ -0,0 +1,62 @@
+import {Result} from '../../api/search/search/result';
+import {Engine} from '../../app/headless-engine';
+import {ResultAnalyticsActions} from '../../features/analytics';
+
+// 1 second is a reasonable amount of time to catch most longpress actions
+const defaultDelay = 1000;
+
+/**
+ * Helpers that more easily allow sending analytics when a search result is selected.
+ */
+export type ResultSelectionHelpers = ReturnType<
+  typeof buildResultSelectionHelpers
+>;
+
+/**
+ * Creates helpers that more easily allow sending analytics when a search result is selected.
+ */
+export function buildResultSelectionHelpers(engine: Engine, result: Result) {
+  let wasOpened = false;
+  const logAnalyticsIfNeverOpened = () => {
+    if (wasOpened) {
+      return;
+    }
+    wasOpened = true;
+    engine.dispatch(ResultAnalyticsActions.logDocumentOpen(result));
+  };
+
+  let longPressTimer: NodeJS.Timeout;
```

Comment:
Hmm, this `NodeJS` type does not look right? It is the same type in all JS run-times?

Reply:
Typescript seems to resolve the return type of `setTimeout` to `NodeJS.Timeout`, I assume this is related to the configuration in `tsconfig.json`.

Although this isn't the *right* run-time, it still transpiles successfully.